Exercise/src/my_string.cpp

```diff
@@ -0,0 +1,67 @@
+#include "my_string.h"
+
+Exercise_1::CMyString::CMyString(const char* charsSequence) 
+{
+	size_t sCurrentIndex = 0;
+	char cCurrentChar = charsSequence[sCurrentIndex];
+	while(cCurrentChar != '\0')
+	{
+		sCurrentIndex++;
+		cCurrentChar = charsSequence[sCurrentIndex];
+	}
+	m_sSize = sCurrentIndex + 1;
+	m_cData = new char[m_sSize];
+
+	
+	for(sCurrentIndex = 0; sCurrentIndex < m_sSize; sCurrentIndex++)
+	{
+		m_cData[sCurrentIndex] = charsSequence[sCurrentIndex];
+	}
+}
+
+// Copy operators
+Exercise_1::CMyString::CMyString(const Exercise_1::CMyString& cmsStringToCopy)
+{
+	// Cleaning data we used before
+	if(m_cData != NULL)
+	{
+		delete[] m_cData;
+	}
+
+	m_sSize = cmsStringToCopy.m_sSize;
+	m_cData = new char[m_sSize];
```

Comment:
What If the old size of m_cData was the same as cmsStringToCopy.m_sSize;?

Reply:
Optimized unneeded allocation and deallocation